test/Application/MarshalMiddlewareTraitTest.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Expressive\Application;
+
+use Interop\Container\ContainerInterface;
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface as ServerMiddlewareInterface;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use ReflectionMethod;
+use ReflectionProperty;
+use stdClass;
+use Zend\Expressive\Application;
+use Zend\Expressive\Exception\InvalidMiddlewareException;
+use Zend\Expressive\Middleware;
+use Zend\Expressive\Router\RouterInterface;
+use Zend\Stratigility\Delegate\CallableDelegateDecorator;
+use Zend\Stratigility\MiddlewarePipe;
+use Zend\Stratigility\Middleware\CallableInteropMiddlewareWrapper;
+use Zend\Stratigility\Middleware\CallableMiddlewareWrapper;
+
+class MarshalMiddlewareTraitTest extends TestCase
```

Comment:
This test covers all behaviors of the `MarshalMiddlewareTrait` following the refactor.